plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelPresenter.java

```diff
@@ -0,0 +1,216 @@
+      GitEventSubscribable subscribeToGitEvents,
+      NotificationManager notificationManager,
+      GitResources gitResources,
+      GitLocalizationConstant locale) {
+    this.view = view;
+    this.service = service;
+    this.changesPanelPresenter = changesPanelPresenter;
+    this.appContext = appContext;
+    this.notificationManager = notificationManager;
+    this.gitResources = gitResources;
+    this.locale = locale;
+
+    this.view.setDelegate(this);
+    this.view.setChangesPanelView(this.changesPanelPresenter.getView());
+
+    this.currentProject = appContext.getRootProject();
+    this.alteredFiles = new MutableAlteredFiles(currentProject);
+
+    if (partStack == null || !partStack.containsPart(this)) {
+      workspaceAgent.openPart(this, PartStackType.NAVIGATION);
+    }
+    eventBus.addHandler(ActivePartChangedEvent.TYPE, this);
+
+    subscribeToGitEvents.addSubscriber(this);
+  }
+
+  /**
+   * Invoked each time when panel is activated.
+   */
+  private void onGitPanelOpen() {
+    // Update file list according to project explorer selection
+    Project selectedProject = appContext.getRootProject();
+
+    if (selectedProject == null) {
+      updateChangedFiles(null);
+      return;
+    } else if (selectedProject != currentProject) {
+      currentProject = selectedProject;
+      reloadChangedFilesList();
+    }
+  }
+
+  private void reloadChangedFilesList() {
+    service
+        .diff(currentProject.getLocation(), null, NAME_STATUS, false, 0, REVISION, false)
+        .then(
+            diff -> {
+              alteredFiles = new MutableAlteredFiles(currentProject, diff);
+              updateChangedFiles(alteredFiles);
+            })
+        .catchError(
+            arg -> {
+              notificationManager.notify(locale.diffFailed(), FAIL, NOT_EMERGE_MODE);
+            });
+  }
+
+  private void updateChangedFiles(AlteredFiles alteredFiles) {
+    changesPanelPresenter.show(alteredFiles);
+  }
+
+  @Override
+  public String getTitle() {
+    return locale.panelTitle();
+  }
+
+  @Override
+  public SVGResource getTitleImage() {
+    return gitResources.git();
+  }
+
+  @Override
+  public IsWidget getView() {
+    return view;
+  }
+
+  @Override
+  public String getTitleToolTip() {
+    return locale.panelTitleToolTip();
+  }
+
+  @Override
+  public void go(AcceptsOneWidget container) {
+    container.setWidget(view);
+  }
+
+  public void showGitPanel() {
+    onActivate();
+  }
+
+  public void hideGitPanel() {
+    partStack.minimize();
+  }
+
+  public boolean isGitPanelOpened() {
+    return partStack.getActivePart() == this;
+  }
+
+  @Override
+  public void onActivePartChanged(ActivePartChangedEvent event) {
+    if (event.getActivePart() != null && event.getActivePart() instanceof GitPanelPresenter) {
+      onGitPanelOpen();
+    }
+  }
+
+  @Override
+  public void onFileChanged(String endpointId, FileChangedEventDto dto) {
+    switch (dto.getStatus()) {
+      case ADDED:
```

Comment:
Why do you need this case segment withot a body?

Reply:
To make the code clearer for understanding. If we have this comments in empty blocks - which probably will be deleted by optimizer - the implementation looks completed and no doubts here that something isn't implemented or forgotten, i.e. the behaviour of this method is designed to do nothing in that cases.